compiler/rustc_ast_lowering/messages.ftl

```diff
@@ -62,6 +60,9 @@ ast_lowering_generic_type_with_parentheses =
     parenthesized type parameters may only be used with a `Fn` trait
     .label = only `Fn` traits may use parentheses
 
+ast_lowering_higher_ranked_gen_param_with_default =
+    higher-ranked generic parameters may not have defaults
```

Comment:
alt: *defaults for {type|const|generic} parameters are not allowed in `for<...>` binders* to mirror *defaults for type parameters are only allowed in `struct`, `enum`, `type`, or `trait` definitions*.

Reply:
sure, i do like that alternative wording better